### PR TITLE
chore: Select component refactoring - SelectControl - Iteration 5

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
The goal of this PR is to remove the instances of the older `react-select` Selects and replace them with the Antdesign Select component. It removes the SelectControl component where possible OR refactors it by mapping the props to minimize the impact.

### BEFORE/AFTER FILTER BOX

https://user-images.githubusercontent.com/60598000/131366035-3ec160f0-a47b-481c-840a-a0d0f96ebc72.mp4

https://user-images.githubusercontent.com/60598000/131366101-b57626d5-791c-4c16-be88-23483bf8c0b7.mp4

### BEFORE/AFTER SPATIAL CONTROL

https://user-images.githubusercontent.com/60598000/131366899-94a6da97-503f-4a28-b277-1393a41e6164.mp4

https://user-images.githubusercontent.com/60598000/131367030-d96a2814-7e0f-44cc-b2ed-0ca786d57b11.mp4

### BEFORE/AFTER ANNOTATION LAYERS

https://user-images.githubusercontent.com/60598000/131367112-ba484ee5-4735-4bed-a0d6-1ebd9b6e2365.mp4

https://user-images.githubusercontent.com/60598000/131367165-bdde3baf-b1b7-4674-969e-fe61708ba1ee.mp4

### BEFORE/AFTER DATASOURCE EDITOR

https://user-images.githubusercontent.com/60598000/131367256-88fa9fdf-c3d4-441c-b43b-9a3330ca2c0b.mp4

https://user-images.githubusercontent.com/60598000/131367315-21ed3bb6-c95f-41b2-b847-1f1798985bb0.mp4

### BEFORE/AFTER CONTROLS

https://user-images.githubusercontent.com/60598000/131532342-929786b5-f6ad-403e-a7d3-1dbc0b27c4ae.mp4

https://user-images.githubusercontent.com/60598000/131532382-7ddcbe82-7597-4cd1-9e04-bd7b33a551f4.mp4

### TESTING INSTRUCTIONS
Test each component individually and all possible controls

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/spec/javascripts/explore/components/SelectControl_spec.jsx [hunk 1]
```diff
@@ -48,58 +48,27 @@ describe('SelectControl', () => {
     wrapper = shallow(<SelectControl {...defaultProps} />);
   });
 
-  it('uses Select in onPasteSelect when freeForm=false', () => {
```
Comment (geido) [inline superset-frontend/spec/javascripts/explore/components/SelectControl_spec.jsx:51]: The OnPasteSelect is not used with the new Select component. These tests are included later checking whether `allowNewOptions` is `true` or `false` based on the SelectControl `freeForm` option.

File: superset-frontend/spec/javascripts/explore/components/SelectControl_spec.jsx [hunk 2]
```diff
@@ -48,58 +48,27 @@ describe('SelectControl', () => {
...
-  });
-
   it('calls props.onChange when select', () => {
     const select = wrapper.instance();
-    select.onChange({ value: 50 });
+    select.onChange(50);
     expect(defaultProps.onChange.calledWith(50)).toBe(true);
   });
 
-  it('returns all options on select all', () => {
```
Comment (geido) [inline superset-frontend/spec/javascripts/explore/components/SelectControl_spec.jsx:69]: The "Select all" option has been removed as agreed with design and all the related tests have been removed as well

File: superset-frontend/spec/javascripts/explore/components/SelectControl_spec.jsx [hunk 3]
```diff
@@ -161,16 +130,6 @@ describe('SelectControl', () => {
         );
         expect(wrapper.html()).not.toContain('add something');
       });
-      it('shows numbers of options as a placeholder by default', () => {
```
Comment (geido) [inline superset-frontend/spec/javascripts/explore/components/SelectControl_spec.jsx:164]: The number of options and remaining options has been removed as agreed with design and all the related tests have been removed as well

File: superset-frontend/spec/javascripts/explore/components/SelectControl_spec.jsx [hunk 4]
```diff
@@ -186,82 +145,12 @@ describe('SelectControl', () => {
...
-      };
-      wrapper.setProps(selectAllProps);
-      expect(wrapper.instance().getOptions(selectAllProps)).toContainEqual({
-        label: 'Select all',
-        meta: true,
-        value: 'Select all',
-      });
-    });
-
-    it('returns the correct options when freeform is set to true', () => {
```
Comment (geido) [inline superset-frontend/spec/javascripts/explore/components/SelectControl_spec.jsx:243]: This test is not necessary with the new Select component as it is a standard behavior

File: superset-frontend/src/explore/components/controls/SelectControl.jsx [hunk 5]
```diff
@@ -18,22 +18,21 @@
...
 import React from 'react';
 import PropTypes from 'prop-types';
-import { t, css } from '@superset-ui/core';
-import { Select, CreatableSelect, OnPasteSelect } from 'src/components/Select';
+import { css } from '@superset-ui/core';
+import { Select } from 'src/components';
 import ControlHeader from 'src/explore/components/ControlHeader';
 
 const propTypes = {
+  ariaLabel: PropTypes.string,
```
Comment (geido) [inline superset-frontend/src/explore/components/controls/SelectControl.jsx:26]: All props here are cleaned and then mapped with the Select ones. This is to minimize impact as the SelectControl component is used dynamically throughout Superset and Superset-ui

File: superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.test.tsx [hunk 6]
```diff
@@ -82,10 +82,22 @@ test('renders extra checkboxes when type is time series', () => {
 });
 
 test('enables apply and ok buttons', async () => {
-  render(<AnnotationLayer {...defaultProps} />);
-  userEvent.type(screen.getByLabelText('Name'), 'Test');
-  userEvent.type(screen.getByLabelText('Formula'), '2x');
-  await waitFor(() => {
+  const { container } = render(<AnnotationLayer {...defaultProps} />);
+
+  waitFor(() => {
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.test.tsx:87]:
```suggestion
  await waitFor(() => {
```

File: superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.test.tsx [hunk 7]
```diff
@@ -82,10 +82,22 @@ test('renders extra checkboxes when type is time series', () => {
...
+  const nameInput = screen.getByRole('textbox', { name: 'Name' });
+  const formulaInput = screen.getByRole('textbox', { name: 'Formula' });
+
+  expect(nameInput).toBeInTheDocument();
+  expect(formulaInput).toBeInTheDocument();
+
+  userEvent.type(nameInput, 'Name');
+  userEvent.type(formulaInput, '2x');
+
+  waitFor(() => {
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.test.tsx:100]:
```suggestion
  await waitFor(() => {
```

**Review by pkdotson — APPROVED** [review]
LGTM! Tested in eph env and everything works great!

File: superset-frontend/src/datasource/DatasourceEditor.jsx [hunk 8]
```diff
@@ -121,7 +121,12 @@ const StyledLabelWrapper = styled.div`
 const checkboxGenerator = (d, onChange) => (
   <CheckboxControl value={d} onChange={onChange} />
 );
-const DATA_TYPES = ['STRING', 'NUMERIC', 'DATETIME', 'BOOLEAN'];
+const DATA_TYPES = [
+  { value: 'STRING', label: 'STRING' },
+  { value: 'NUMERIC', label: 'NUMERIC' },
+  { value: 'DATETIME', label: 'DATETIME' },
+  { value: 'BOOLEAN', label: 'BOOLEAN' },
+];
```
Comment (villebro) [inline superset-frontend/src/datasource/DatasourceEditor.jsx:129]: Not for this PR, but this probably shouldn't be a Select component at all, but rather a text field. These should be raw column types coming from the database, hence a typical `STRING` will be `VARCHAR(x)`, `TEXT` etc.

**Review by villebro — APPROVED** [review]
This is amazing work, comparing the before/after was such a visible improvement! 🎉  LGTM!

**Pull request merged**